pandas/core/arrays/interval.py

```diff
@@ -1750,5 +1753,6 @@ def _maybe_convert_platform_interval(values) -> ArrayLike:
         values = extract_array(values, extract_numpy=True)
 
     if not hasattr(values, "dtype"):
-        return np.asarray(values)
```

Comment:
why is this change needed?

Reply:
It may look like it's not needed. Maybe it was something I added before #50130. I'll try removing it again.